datumaro/util/os_util.py

```diff
@@ -49,6 +62,54 @@ def walk(path, max_depth=None):
 
         yield dirpath, dirnames, filenames
 
+def copytree(src, dst):
+    # Shutil works very slow pre 3.8
+    # https://docs.python.org/3/library/shutil.html#platform-dependent-efficient-copy-operations
+    # https://bugs.python.org/issue33671
+
+    if sys.version_info >= (3, 8):
+        shutil.copytree(src, dst)
+        return
+
+    basedir = osp.dirname(dst)
+    if basedir:
+        os.makedirs(basedir, exist_ok=True)
+
+    if sys.platform == 'windows':
+        # Ignore
+        #   B603: subprocess_without_shell_equals_true
+        #   B607: start_process_with_partial_path
+        # In this case we control what is called and command arguments
+        # PATH overriding is considered low risk
+        assert src and dst
+        src = osp.abspath(src)
+        dst = osp.abspath(dst)
+        subprocess.check_output(["xcopy", src, dst, # nosec
+            "/s", "/e", "/q", "/y", "/i"], stderr=subprocess.STDOUT) # nosec
+    elif sys.platform == 'linux':
+        # As above
+        subprocess.check_call(["cp", "-r", src, dst]) # nosec
+    else:
+        shutil.copytree(src, dst)
+
+@contextmanager
+def suppress_output(stdout: bool = True, stderr: bool = False):
+    with open(os.devnull, 'w') as devnull, ExitStack() as es:
+        if stdout:
+            es.enter_context(redirect_stdout(devnull))
+        elif stderr:
+            es.enter_context(redirect_stderr(devnull))
+        with es:
```

Comment:
This is a duplicate `with` - you have already entered the `es` context on the first line.

Reply:
Fixed